pylint/test/functional/unused_import_type_hint_py3.py

```diff
@@ -0,0 +1,12 @@
+"""Test of unused import warnings while type hinting"""
+# pylint: disable=wrong-import-position, unused-argument, import-error, missing-docstring
+from fake import SomeName  # [unused-import]
+
+# following should be ok
+from fake import OtherName
+def my_method(attribute: OtherName):
+    pass
+
+from fake import AnotherName
+def another_method(attribute: 'AnotherName'):
+    pass
```

Comment:
We need additional tests here, e.g. when used as `List[AnotherName]`, `Optional` or any other 'composite' type
e.g. [with type aliases](https://docs.python.org/3/library/typing.html#type-aliases)

Maybe more...